src/osm2pgsql-gen.cpp

```diff
@@ -0,0 +1,365 @@
+            case 'G':
+                params["group_by_column"] = optarg;
+                break;
+            case 'l':
+                log_level = optarg;
+                break;
+            case 'm':
+                params["margin"] = optarg;
+                break;
+            case 'p':
+                add_param(&params, optarg);
+                break;
+            case 's':
+                strategy = optarg;
+                break;
+            case 't':
+                params["src_table"] = optarg;
+                break;
+            case 'T':
+                params["dest_table"] = optarg;
+                break;
+            case 'w':
+                params["image_extent"] = optarg;
+                break;
+            case 'x':
+                x = parse_uint32(optarg, "x");
+                break;
+            case 'y':
+                y = parse_uint32(optarg, "y");
+                break;
+            case 'X':
+                xmax = parse_uint32(optarg, "X");
+                break;
+            case 'Y':
+                ymax = parse_uint32(optarg, "Y");
+                break;
+            case 'z':
+                zoom = parse_uint32(optarg, "zoom");
+                break;
+            case 200:
+                params["img_path"] = optarg;
+                break;
+            case 201:
+                params["img_table"] = optarg;
+                break;
+            case 202:
+                get_logger().enable_sql();
+                break;
+            default:
+                log_error("Unknown argument");
+                return 2;
+            }
+        }
+
+        if (params.count("src_table") == 0) {
+            log_error("Missing required -t, --table option");
+            return 2;
+        }
+
+        if (params.count("dest_table") == 0) {
+            log_error("Missing required -T, --dest-table option");
+            return 2;
+        }
+
+        if (log_level == "debug") {
+            get_logger().set_level(log_level::debug);
+        } else if (log_level == "info" || log_level.empty()) {
+            get_logger().set_level(log_level::info);
+        } else if (log_level == "warn") {
+            get_logger().set_level(log_level::warn);
+        } else if (log_level == "error") {
+            get_logger().set_level(log_level::error);
+        } else {
+            log_error(
+                "Unknown log level: {}. "
+                "Use 'debug', 'info', 'warn', or 'error'."_format(log_level));
+            return 2;
+        }
+
+        util::timer_t timer_overall;
+
+        log_info("osm2pgsql-gen version {}", get_osm2pgsql_version());
+        pg_conn_t conn{""};
+
+        if (params.count("expire_list")) {
+            log_info("Get tiles to reprocess from table '{}'",
+                     params["expire_list"]);
+            x = 0;
+            y = 0;
+            xmax = 0;
+            ymax = 0;
+        } else {
+            if (x == 0 && y == 0 && xmax == 0 && ymax == 0) {
```

Comment:
Zero shouldn't be treated as a special value, because you might just want to generate the 0/0 tile.

Reply:
fixed in new version